src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1713,13 +1693,10 @@ void Propagate(DependencyObjectStore store)
 				for (var propertyIndex = 0; propertyIndex < props.Length; propertyIndex++)
 				{
 					var prop = props[propertyIndex];
-					if (!IsTemplatedParentFrozen || prop != TemplatedParentProperty)
+					var precedence = GetCurrentHighestValuePrecedence(prop);
+					if (precedence is not DependencyPropertyValuePrecedences.DefaultValue)
 					{
-						var value = GetValue(prop);
-						if (GetCurrentHighestValuePrecedence(prop) != DependencyPropertyValuePrecedences.DefaultValue)
-						{
-							store.OnParentPropertyChangedCallback(instanceRef, prop, value);
-						}
+						store.OnParentPropertyChangedCallback(instanceRef, prop, GetValue(prop, precedence));
```

Comment:
GetValue call needs to be unconditional as it was previously. It has the side effect of TryRegisterInheritedProperties that we are currently relying on (yes, it's weird)

Reply:
ok... i optimized it because we will be reading the precedence twice
guess that was messed up